src/components/Calculator.js

```diff
@@ -1,42 +1,69 @@
 import React, { Component } from 'react';
 import Button from './Button';
+import calculate from '../logic/calculate';
 
 class Calculator extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      operation: null,
+      total: 0,
+      next: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
   }
 
-  render() {
-    return (
-      <div className="wrapper">
-        <div className="calculator">
-          <div className="screen">0</div>
-        </div>
-        <div className="keyboard">
-          <Button>AC</Button>
-          <Button>+/-</Button>
-          <Button>%</Button>
-          <Button className="orange">÷</Button>
-          <Button>7</Button>
-          <Button>8</Button>
-          <Button>9</Button>
-          <Button className="orange">x</Button>
-          <Button>4</Button>
-          <Button>5</Button>
-          <Button>6</Button>
-          <Button className="orange">-</Button>
-          <Button>1</Button>
-          <Button>2</Button>
-          <Button>3</Button>
-          <Button className="orange">+</Button>
-          <Button id="zero">0</Button>
-          <Button>.</Button>
-          <Button className="orange">=</Button>
+handleClick = (e) => {
+  this.setState((state) => calculate(state, e.target.innerText));
+}
+
+render() {
+  const { total, next, operation } = this.state;
+  return (
+    <div className="wrapper">
+      <div className="calculator">
+        <div className="screen">
+          {total}
+          {next}
+          {operation}
         </div>
```

Comment:
- [ ] Great job! Your calculator is giving us the correct answers, but there is an issue with the way the digits are displayed, when you enter the first number, press an operator, as you are entering the second number, the digits are concatenating with the first number, please kindly fix the issue. Because normally when you enter the first number, press an operator, by the time you are entering the second number, the calculator should either display the second number after the operator or it should clear the first number but for your case it is joining the first and second number before the operator, please fix that. `123 + 456 is displaying as 123456+`

Reply:
I will fix that. Thank you very much